components/sup/src/main.rs

```diff
@@ -202,76 +202,85 @@ fn sub_term(m: &ArgMatches) -> Result<()> {
-    let mut cfg = ManagerConfig::default();
-    cfg.auto_update = m.is_present("AUTO_UPDATE");
-    cfg.update_url = bldr_url(m);
-    cfg.update_channel = channel(m);
+    let mut cfg = ManagerConfig {
+        auto_update: m.is_present("AUTO_UPDATE"),
+        update_url: bldr_url(m),
+        update_channel: channel(m),
+        http_disable: m.is_present("HTTP_DISABLE"),
+        organization: m.value_of("ORGANIZATION").map(|org| org.to_string()),
+        gossip_permanent: m.is_present("PERMANENT_PEER"),
+        ring_key: get_ring_key(m)?,
+        ..Default::default()
+    };
     if let Some(addr_str) = m.value_of("LISTEN_GOSSIP") {
         cfg.gossip_listen = GossipListenAddr::from_str(addr_str)?;
     }
     if let Some(addr_str) = m.value_of("LISTEN_HTTP") {
         cfg.http_listen = http_gateway::ListenAddr::from_str(addr_str)?;
     }
-    cfg.http_disable = m.is_present("HTTP_DISABLE");
     if let Some(addr_str) = m.value_of("LISTEN_CTL") {
-        cfg.ctl_listen =
-            SocketAddr::from_str(addr_str).unwrap_or_else(|_err| protocol::ctl::default_addr());
+        cfg.ctl_listen = SocketAddr::from_str(addr_str)?;
     }
-    cfg.organization = m.value_of("ORGANIZATION").map(|org| org.to_string());
-    cfg.gossip_permanent = m.is_present("PERMANENT_PEER");
-    // TODO fn: Clean this up--using a for loop doesn't feel good however an iterator was
-    // causing a lot of developer/compiler type confusion
-    let mut gossip_peers: Vec<SocketAddr> = Vec::new();
     if let Some(peers) = m.values_of("PEER") {
-        for peer in peers {
-            let peer_addr = if peer.find(':').is_some() {
-                peer.to_string()
-            } else {
-                format!("{}:{}", peer, GOSSIP_DEFAULT_PORT)
-            };
-            let addrs: Vec<SocketAddr> = match peer_addr.to_socket_addrs() {
-                Ok(addrs) => addrs.collect(),
-                Err(e) => {
-                    outputln!("Failed to resolve peer: {}", peer_addr);
-                    return Err(sup_error!(Error::NameLookup(e)));
-                }
-            };
-            let addr: SocketAddr = addrs[0];
-            gossip_peers.push(addr);
-        }
+        cfg.gossip_peers = get_peers(peers)?;
     }
-    cfg.gossip_peers = gossip_peers;
     if let Some(watch_peer_file) = m.value_of("PEER_WATCH_FILE") {
         cfg.watch_peer_file = Some(String::from(watch_peer_file));
     }
-    cfg.ring_key = match m.value_of("RING") {
-        Some(val) => Some(SymKey::get_latest_pair_for(
-            &val,
-            &default_cache_key_path(None),
-        )?),
+    if let Some(events) = m.value_of("EVENTS") {
+        cfg.eventsrv_group = ServiceGroup::from_str(events).ok().map(Into::into);
+    }
+    Ok(cfg)
+}
+
+// Various CLI Parsing Functions
+////////////////////////////////////////////////////////////////////////
+
+fn get_peers(peers: clap::Values) -> Result<Vec<SocketAddr>> {
```

Comment:
I realize you're just moving existing code into a function, but at long as it's getting mutated, we might take this opportunity of fix some things up.

First, why not have `get_peers` take `m: &ArgMatches` similar to `get_ring_key`? Add an outer condition like so:
```rust
    if let Some(peers) = m.values_of("PEER") {
        for peer in peers {
            …
        }
    }
```

and you can get the same simplicity of passing `gossip_peers: get_peers(m)?` directly to the `ManagerConfig` initializer.